app/summary/page.tsx

```diff
@@ -0,0 +1,143 @@
+"use client";
+
+import untypedSubjectSummaries from "@/app/generated/subject-summaries.json";
+import { SummaryRow } from "@/app/types";
+import { useState, useRef, useEffect } from "react";
+
+const SUMMARY_DISPLAY_COUNT = 100;
+const subjectSummaries = untypedSubjectSummaries as unknown as SummaryRow[];
+const tableHeaders = [
+    {
+        key: "enrollmentTerm",
+        label: "Term"
+    },
+    {
+        key: "subjectArea",
+        label: "Subject"
+    },
+    {
+        key: "catalogNumber",
+        label: "Course Number"
+    },
+    {
+        key: "courseTitle",
+        label: "Course Name"
+    },
+    {
+        key: "instructorName",
+        label: "Professor"
+    },
+    {
+        key: "gradeAverage",
+        label: "Average Grade"
+    },
+    {
+        key: "gradeSpread",
+        label: "Grade Spread"
+    },
+    {
+        key: "gradedTotal",
+        label: "Total Graded"
+    },
+    {
+        key: "enrollmentTotal",
+        label: "Total Enrolled"
+    }
+]
+
+const Summary = () => {
+    const [cutoff, setCutoff] = useState(SUMMARY_DISPLAY_COUNT);
+    const [sortBy, setSortBy] = useState<keyof SummaryRow>("gradeAverage");
+    const [sortReverse, setSortReverse] = useState(true);
+    const [displaySummaries, setDisplaySummaries] = useState(subjectSummaries.slice(0, cutoff));
+    const observerRef = useRef<HTMLDivElement>(null);
+
+    const onHeaderClick = (key: keyof SummaryRow) => {
+        if (sortBy === key) {
+            setSortReverse(!sortReverse);
+        } else {
+            setSortBy(key);
+            setSortReverse(false);
+        }
+        setCutoff(SUMMARY_DISPLAY_COUNT);
+    };
+
+    useEffect(() => {
+        const observer = new IntersectionObserver((entries) => {
+            if (entries[0].isIntersecting) {
+                setCutoff(c => c + SUMMARY_DISPLAY_COUNT);
+            }
+        });
+
+        const node = observerRef.current;
+
+        if (node) {
+            observer.observe(node);
+        }
+
+        return () => {
+            if (node) {
+                observer.unobserve(node);
+            }
+        }
+    }, [observerRef]);
```

Comment:
nice infinite scroll implementation! it might also be worthwhile to consider using like [react-virtualized](https://github.com/bvaughn/react-virtualized) to hide rows when the user has already scrolled past them